cli/EAP7-943_Security_Auth_Commands.adoc

```diff
@@ -0,0 +1,241 @@
+ 
+This command removes the active HTTP factory or a mechanism from the active HTTP factory. 
+Without any mechanism provided, this command erases the factory from the interface. 
+NB: Existing resources are not removed from management model. +
+ 
+_security disable-http-auth-management [--mechanism=<a string>]
+                                          [--no-reload]_
+
+==== Enabling/Disabling HTTP Authentication for undertow security domain
+ 
+==== enable-http-auth-http-server
+
+Associate an elytron HTTP factory to the security domain. In case a SASL factory already exists, 
+the factory is extended(or updated) with the selected mechanism. 
+If no mechanism is provided, the Out of The Box Application HTTP factory is 
+associated to the security domain. This command creates all the non existing 
+resources required to configure authentication. This command can be called multiple 
+times to configure the referenced HTTP factory. Similar behaviour than _enable-http-auth-management_ command. +
+ 
+_security enable-http-auth-http-server --security-domain=<a string> (
+                                          [--file-system-realm=<a string>]
+                                          [--user-role-decoder=<a string>] |
+                                          [--user-properties-file=<null>]
+                                          [--group-properties-file=<null>]
+                                          [--relative-to=<a string>] )
+                                          [--mechanism=<a string>]
+                                          [--exposed-realm-name=<a string>]
+                                          [--external-trust-store=<a string>]
+                                          [--new-auth-factory-name=<a string>]
+                                          [--new-realm-name=<a string>]
+                                          [--new-security-domain-name=<a
+                                          string>] [--no-reload]_
+ 
+ 
+The mininal command to associate the out of the box elytron HTTP factory to the Foo security-domain: +
+_security enable-http-auth-http-server --security-domain=Foo_
+ 
+==== Options
+
+--security-domain=Mandatory, the undertow security domain. +
+Other options are the same than enable-sasl-management. The completer for _--mechanism_ exposes HTTP mechanisms.
+ 
+==== disable-http-auth-http-server
+ 
+This command removes the security domain or a mechanism from the active HTTP factory. 
+Without a mechanism, the security domain is removed. 
+NB: Elytron existing resources are not removed from management model.+
+ 
+_security disable-http-auth-http-server --security-domain=<a string>
+                                           [--mechanism=<a string>]
+                                           [--no-reload]_
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+No support for mechanisms not listed in Hard Requirements. +
+These actions are not available in domain mode.
```

Comment:
Analysis could contains the reason for this. This should be also in documentation.

AFAIK (thanks to @mchoma's explanation), CLI can't support domain, because Elytron doesn't support domain. And Elytron can't support domain because Domain doesn't provide mechanisms for keystore distribution to domain servers ...

Reply:
It's unclear to me what effects these commands have on the target process, beyond the obvious one of changing the management model and its persisted xml config file. For example, is the target process actually creating the properties files if the command params specify those, or is this just a case of manipulating config to reference files that already exist?

I think the analysis should clarify this regardless, but I raise it here because it relates to the domain mode question @marekkopecky raised.

The xxx-management commands are not applicable to a domain overall, as there are no relevant domain-wide resources. They could apply to a HostController though. An HC is analogous to a standalone server. There's no question of distributing files to multiple hosts in the domain, as the command targets just one.

The xxx-http-server commands could work fine in a domain, so long as none of the commands are meant to change things outside the config model (e.g. create properties files, which would require non-existent distribution support.)

There are other things that would have to be handled for domain mode support, e.g. the user specifying what profile the xxx-http-server commands relate to, and what host the xxx-management commands relate to. 

I also think it's fine to say domain mode isn't covered just because it's out of scope for this version of the Feature but *may* be considered in the future. 

Anyway, I think some comment about why domain mode isn't covered should be added.

Reply:
That is good point. 

It seems to me these commands are meant to rely on existing artifacts. Otherwise they are missing many parameters. In origin document there was explicitly stated <cite>Adding users to properties files or file-system based is out of the scope of the following commands.<cite>

I also believe host for the xxx-management could be find out automatically and user does not have to provide this information.